File: src/infrastructure/servers/factories/langchain_server_factory.py
from langchain_openai import ChatOpenAI
from pydantic import SecretStr
from infrastructure.servers.factories.abstracts.base_server_factory import BaseServerFactory

class LangChainServerFactory(BaseServerFactory):

    def build_server(self) -> ChatOpenAI:
        self.model: str = "llama3-70b-8192"
        api_key: SecretStr = SecretStr(secret_value=self.require_env(
            key="LLM__GROQ_API_KEY",
            error_message="LangChain API key não encontrada no .env"))

        timeout: float = self.optional_float_env(
            key="LLM__TIMEOUT_SECONDS", default=30.0
        )

        max_retries: int = self.optional_int_env(key="LLM__MAX_RETRIES", default=1)

        return ChatOpenAI(
            api_key=api_key, model=self.model, timeout=timeout, max_retries=max_retries
        )

File: angular-app/src/app/core/infrastructure/servers/factories/langchain-server-factory.ts
import { IServer, TextCompletionResponse } from '../abstracts/i-server';
import { BaseServerFactory } from './abstracts/base-server-factory';

export class LangChainServerFactory extends BaseServerFactory {
  buildServer(): IServer {
    const model: string = 'llama3-70b-8192';
    const apiKey: string = this.requireEnv('LLM__GROQ_API_KEY', 'LangChain API key não encontrada no .env');
    const timeout: number = this.optionalFloatEnv('LLM__TIMEOUT_SECONDS', 30.0);
    const maxRetries: number = this.optionalIntEnv('LLM__MAX_RETRIES', 1);

    return {
      invoke: async (): Promise<TextCompletionResponse> => ({
        model,
        text: `[mock:${apiKey}:${timeout}:${maxRetries}]`,
      }),
      models: {
        list: () => ({
          data: [{ id: model }],
        }),
      },
    };
  }
}
